qiskit_experiments/framework/experiment_data.py

```diff
@@ -12,93 +12,1803 @@
+    3. Saving/Loading data from the result database
+    """
+
+    _metadata_version = 1
+    _job_executor = futures.ThreadPoolExecutor()
+
+    _json_encoder = ExperimentEncoder
+    _json_decoder = ExperimentDecoder
+
+    _metadata_filename = "metadata.json"
+
+    def __init__(
+        self,
+        experiment: Optional["BaseExperiment"] = None,
+        backend: Optional[Backend] = None,
+        service: Optional[IBMExperimentService] = None,
+        parent_id: Optional[str] = None,
+        job_ids: Optional[List[str]] = None,
+        child_data: Optional[List[ExperimentData]] = None,
+        verbose: Optional[bool] = True,
+        db_data: Optional[ExperimentDataclass] = None,
+        **kwargs,
+    ):
+        """Initialize experiment data.
+
+        Args:
+            experiment: Optional, experiment object that generated the data.
+            backend: Optional, Backend the experiment runs on.
+            service: The service that stores the experiment results to the database
+            parent_id: Optional, ID of the parent experiment data
+                in the setting of a composite experiment
+            job_ids: Optional, IDs of jobs submitted for the experiment.
+            child_data: Optional, list of child experiment data.
+            verbose: Optional, whether to print messages
+            db_data: Optional, a prepared ExperimentDataclass of the experiment info;
+            overrides other db parameters.
+        """
+        if experiment is not None:
+            backend = backend or experiment.backend
+            experiment_type = experiment.experiment_type
+        else:
+            experiment_type = None
+        if job_ids is None:
+            job_ids = []
+
+        self._experiment = experiment
+
+        # data stored in the database
+        metadata = dict()
+        if experiment is not None:
+            metadata = copy.deepcopy(experiment._metadata())
+        source = metadata.pop(
+            "_source",
+            {
+                "class": f"{self.__class__.__module__}.{self.__class__.__name__}",
+                "metadata_version": self.__class__._metadata_version,
+                "qiskit_version": qiskit_version(),
+            },
+        )
+        metadata["_source"] = source
+        experiment_id = kwargs.get("experiment_id", str(uuid.uuid4()))
+        if db_data is None:
+            self._db_data = ExperimentDataclass(
+                experiment_id=experiment_id,
+                experiment_type=experiment_type,
+                parent_id=parent_id,
+                job_ids=job_ids,
+                metadata=metadata,
+            )
+        else:
+            self._db_data = db_data
+
+        for key, value in kwargs.items():
+            if hasattr(self._db_data, key):
+                setattr(self._db_data, key, value)
+
+        # general data related
+        self._backend = None
+        if backend is not None:
+            self._set_backend(backend)
```

Comment:
It might work for `None` in this case, but it's not supposed to run on `None`. Also, it will set the backend name in the db data to "None" but it should not set it.

Reply:
I need to check before `get_service_from_backend` as well. Added.